adapter.go

```diff
@@ -83,49 +103,51 @@ func NewFilteredAdapter(url string) persist.FilteredAdapter {
 }
 
 func (a *adapter) open() {
-	// FailFast will cause connection and query attempts to fail faster when
-	// the server is unavailable, instead of retrying until the configured
-	// timeout period. Note that an unavailable server may silently drop
-	// packets instead of rejecting them, in which case it's impossible to
-	// distinguish it from a slow server, so the timeout stays relevant.
-	a.dialInfo.FailFast = true
+	ctx, cancle := context.WithTimeout(context.TODO(), a.timeout)
+	defer cancle()
 
-	if a.dialInfo.Database == "" {
-		a.dialInfo.Database = "casbin"
-	}
-
-	session, err := mgo.DialWithInfo(a.dialInfo)
+	client, err := mongo.Connect(ctx, a.clientOption)
 	if err != nil {
 		panic(err)
 	}
 
-	db := session.DB(a.dialInfo.Database)
-	collection := db.C("casbin_rule")
+	db := client.Database("casbin")
```

Comment:
We should allow users to customize the databaseName.

Reply:
This is a bit difficult, We must add a `databaseName` parameter to the function,I think such a function is a bit less elegant. Many users may not want to specify this name. Or we can provide another function to allow users to create the db themselves
```
func NewAdapterWithDatabase (database * mongo.Database, timeout ... interface ()) persist.Adapter
```
Which one do you think is better?

Reply:
See README about how to provide DB: https://github.com/casbin/mongodb-adapter#simple-example

```go
	// Or you can use an existing DB "abc" like this:
	// The adapter will use the table named "casbin_rule".
	// If it doesn't exist, the adapter will create it automatically.
	// a := mongodbadapter.NewAdapter("127.0.0.1:27017/abc")
```

Reply:
Thank you for your reminder